extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/RequestScopeHandlerInterceptor.java

```diff
@@ -43,7 +48,7 @@ public void close(Status status, Metadata trailers) {
                         capturedVertxContext.runOnContext(new Handler<Void>() {
                             @Override
                             public void handle(Void ignored) {
-                                reqContext.deactivate();
+                                reqContext.destroy(finalState);
```

Comment:
For my own curiosity, what's the difference?

Reply:
`deactivate()` merely removes the state from a threadlocal variable. `destroy()` destroys all the bean instances correctly.

Reply:
I know it's complicated but we need this in order to be able to propagate the context and follow the CDI rules for normal scopes...

Reply:
maybe we could rename the methods to shutdown and clear? I'm not sure it would be more clear to anyone but me :D

Reply:
> maybe we could rename the methods to shutdown and clear? I'm not sure it would be more clear to anyone but me :D

-1 the method names somehow relate to the CDI API, e.g. `Context.isActive()` and `Contextual.destroy()`.